pkg/operator/sync_openshiftcontrollermanager_v311_00_test.go

```diff
@@ -115,15 +116,16 @@ func TestConfigMapControllerDisabling(t *testing.T) {
 			operatorConfigClient: controllerManagerOperatorClient.OperatorV1(),
 			clusterVersionLister: clusterVersionLister,
 		}
+		result := map[string][]string{}
 		resultConfigMap, _, err := manageOpenShiftControllerManagerConfigMap_v311_00_to_latest(clusterVersionLister, kubeClient, configMapsGetter, operator.recorder, operatorConfig)
 		if err != nil {
 			t.Error(err)
+		} else {
+			json.Unmarshal([]byte(resultConfigMap.Data["config.yaml"]), &result)
 		}
-		result := map[string][]string{}
-		json.Unmarshal([]byte(resultConfigMap.Data["config.yaml"]), &result)
+		sort.Strings(result["controllers"])
 		resultControllers := map[string][]string{"controllers": result["controllers"]}
-
-		if !reflect.DeepEqual(tc.result, resultControllers) {
+		if err == nil && !reflect.DeepEqual(tc.result, resultControllers) {
```

Comment:
Given resourcemerge.MergeConfigMap overlays individual config maps we should check the whole configuration. Similar to https://github.com/openshift/cluster-openshift-apiserver-operator/pull/532/commits/2cc54289a7b0e7e973311726a4723b80f01e18db#diff-25e7c9a2521d15c93e122ab635c1ec19358116650aa275abca5e014b514c69c2R368-R399 to avoid the configuration changing in an unexpected way.

Reply:
I prefer to leave the tests that are looking at the generated string slice for controllers to be looking specifically at that.

I added a dedicated test for the full generated config with a single testcase: https://github.com/openshift/cluster-openshift-controller-manager-operator/pull/291/commits/3d6df03ad2f53e6d3b06122c4cfdca13fbb4ef10